### PR TITLE
Feature: show status

### LINKED ISSUE #52: Show response status and duration

### LINKED ISSUE #227: Show status  for the http code returned on success and on fail
**Why have this feature?**
This feature will help drive clarity to the user on what is happening in the background and communicate the results of the query. 

**Describe the solution you'd like**
Show status  for the http code returned on success and on fail  as well with the right background color and the right **Icon** as in the screenshots below: 
image

image

Related to #52

### PR DESCRIPTION
**Pull request by cwahome**
## Overview

Shows the error/success message and duration of the query

Fixes #227 and for subsequently Fixes #52 

### Demo

Successful request:
image

Failed request
image

## Testing Instructions

* Run the query already on the runner and view the message bar for a success message
* Edit the query on the runner by adding 'd' after the slash, run the query and view the message bar for a failure message

### CODE REVIEW

File: src/app/services/reducers/query-runner-reducers.ts [hunk 1]
```diff
@@ -8,8 +8,8 @@ export function graphResponse(state = {}, action: IAction): any {
       return action.response;
     case VIEW_HISTORY_ITEM_SUCCESS:
       return action.response;
-    case QUERY_GRAPH_ERROR:
-      return { body: {}, headers: {} };
+    // case QUERY_GRAPH_STATUS:
```
Comment (andrueastman) [inline src/app/services/reducers/query-runner-reducers.ts]: Do you intend to have this commented out?

File: src/tests/services/actions/query-action-creators.spec.tsx [hunk 2]
```diff
@@ -33,8 +33,6 @@ describe('query actions', () => {
           body: undefined,
           createdAt,
           duration: undefined,
-          // tslint:disable-next-line:max-line-length
```
Comment (andrueastman) [inline src/tests/services/actions/query-action-creators.spec.tsx]: 👍

File: src/app/services/reducers/query-runner-reducers.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import { IAction } from '../../../types/action';
-import { CLEAR_RESPONSE, QUERY_GRAPH_ERROR, QUERY_GRAPH_SUCCESS,
+import { CLEAR_RESPONSE, QUERY_GRAPH_STATUS, QUERY_GRAPH_SUCCESS,
```
Comment (andrueastman) [inline src/app/services/reducers/query-runner-reducers.ts]: Missed a spot. Please remove the unused import

File: src/app/services/reducers/query-loading-reducers.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import { IAction } from '../../../types/action';
 import {
   FETCH_SCOPES_ERROR, GET_CONSENT_ERROR,
```
Comment (jobala) [inline src/app/services/reducers/query-loading-reducers.ts]: Formatting the imports like shown below makes it easy to read.
```
FETCH_SCOPES_ERROR,
GET_CONSENT_ERROR
```

**Comment by jobala** [comment]
####Bug
image

The logic for calculating duration is duplicated between permission and query action creators, perhaps abstract into a function?

**Comment by cwahome** [comment]
> ####Bug
> image
> 
> The logic for calculating duration is duplicated between permission and query action creators, perhaps abstract into a function?

Nice to have but the logic as is is negligible

**Pull request merged**